.github/workflows/publish-release.yml

```
@@ -38,6 +40,7 @@ jobs:
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
       - run: yarn --immutable
+      - run: echo "INFURA_PROJECT_ID=${{ secrets.INFURA_PROJECT_ID }}" > .env
       - run: yarn build
```

Can't you do something like this?
```suggestion
      - run: yarn build
        env:
          INFURA_PROJECT_ID: ${{ secrets.INFURA_PROJECT_ID }}
```

---

I can, but in my past attempts I found that .env was a safer option, as ENV vars wouldn't always get picked up in scripts. It's also possible I had a different problem back then, so I will apply your suggestion to the rest of the workflows as well

---

I applied your suggestion in #40